### PR TITLE
chore: implement infra nginx. (MAPCO-3391)

### PR DESCRIPTION
**Pull request by almog8k**

### CODE REVIEW

File: helm/config/nginx.conf [hunk 1]
```diff
@@ -1,117 +1,111 @@
...
+            '"proxy_protocol_server_port": "$proxy_protocol_server_port",'
+            '"proxy_host": "$proxy_host",'
+            '"proxy_port": "$proxy_port",'
+            '"realip_remote_addr": "$realip_remote_addr",'
+            '"realip_remote_port": "$realip_remote_port",'
+            '"server_name": "$server_name",'
+            '"server_addr ": "$server_addr ",'
+            '"server_port": "$server_port",'
+            '"server_protocol": "$server_protocol",'
+            '"ssl_client_serial": "$ssl_client_serial",'
```
Comment (asafMasa) [inline helm/config/nginx.conf:71]: remove ssl logs

File: helm/templates/configmap-nginx.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-nginx-configmap
+data:
+  default.conf: {{ tpl (.Files.Get "config/default.conf") . | quote }}
+  nginx.conf: {{ tpl (.Files.Get "config/nginx.conf") . | quote }}
+  nginx.mapproxy.conf: {{ tpl (.Files.Get "config/nginx.mapproxy.conf") . | quote }}
+  nginx.mapproxy.wmts.conf: {{ tpl (.Files.Get "config/nginx.mapproxy.wmts.conf") . | quote }}
+{{- end }}
```
Comment (asafMasa) [inline helm/templates/configmap-nginx.yaml:11]: add end of file space

File: helm/templates/mapproxinator-container.yaml [hunk 3]
```diff
@@ -80,7 +80,7 @@
             - mountPath: "/etc/ssl/openssl.cnf"
               name: open-ssl
               subPath: "openssl.cnf"
-            {{- end }}
+            {{- end }}
```
Comment (asafMasa) [inline helm/templates/mapproxinator-container.yaml:83]: revert space

File: helm/templates/mapproxy-bundle.yaml [hunk 4]
```diff
@@ -75,7 +75,7 @@
         - name: open-ssl
           secret:
             secretName: {{ $sslSecretName }}
-        {{- end }}
+        {{- end }}
```
Comment (asafMasa) [inline helm/templates/mapproxy-bundle.yaml:78]: revert space

File: helm/templates/mapproxy-init-container.yaml [hunk 5]
```diff
@@ -42,7 +42,7 @@
             {{- if .Values.rasterCommon.ca.secretName }}
             - name: REQUESTS_CA_BUNDLE
               value: '/usr/local/share/ca-certificates/ca.crt'
-            {{- end }}
+            {{- end }}
```
Comment (asafMasa) [inline helm/templates/mapproxy-init-container.yaml:45]: revert space

File: helm/templates/mapproxy-init-container.yaml [hunk 6]
```diff
@@ -65,7 +65,7 @@
             - mountPath: "/etc/ssl/openssl.cnf"
               name: open-ssl
               subPath: "openssl.cnf"
-            {{- end }}
+            {{- end }}
```
Comment (asafMasa) [inline helm/templates/mapproxy-init-container.yaml:68]: revert spaces

**Review by asafMasa — CHANGES_REQUESTED** [review]
Good job

File: helm/config/default.conf [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+# Define upstreams for WMS and WMTS respectively
+upstream mapproxy-wms {
+    server {{ $serviceName }}-wms:{{ .Values.mapproxy.uwsgi.socket }};
+}
+
+upstream mapproxy-wmts {
+    server {{ $serviceName }}-wmts:{{ .Values.mapproxy.uwsgi.socket }};
+}
+
+map $msec $nanosec {
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply helm/config/default.conf:26]: as i know the opa-la repo we override this nginx config file by 'extraVolumeMounts', in that case it is necessary

File: helm/config/nginx.conf (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply helm/config/nginx.conf:71]: removed all SSL logs

File: helm/templates/mapproxy-bundle.yaml (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/mapproxy-bundle.yaml:78]: fixed

File: helm/templates/mapproxinator-container.yaml (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/mapproxinator-container.yaml:83]: fixed

File: helm/templates/mapproxy-init-container.yaml (hunk 5 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/mapproxy-init-container.yaml:45]: fixed

File: helm/templates/mapproxy-init-container.yaml (hunk 6 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/mapproxy-init-container.yaml:68]: fixed

File: helm/templates/configmap-nginx.yaml (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/configmap-nginx.yaml]: fixed

**Review by asafMasa — APPROVED** [review]

**Pull request merged**